### PR TITLE
Stripe Credit Card Payments

### PR DESCRIPTION
**Pull request by ppj**
Offer to receive payments during sign-up
Process payments directly through Stripe

### CODE REVIEW

File: app/views/users/new.html.haml [hunk 1]
```diff
@@ -9,6 +15,30 @@
...
+        %fieldset.credit_card
+          %span.payment-errors
+          .form-group
+            = label_tag nil, "Credit Card Number", class: "control-label col-sm-2"
+            .col-sm-6
+              = text_field_tag nil, nil, class: "form-control", data: {stripe: "number"}
+          .form-group
+            = label_tag nil, "Security Code", class: "control-label col-sm-2"
+            .col-sm-6
+              = text_field_tag nil, nil, class: "form-control", data: {stripe: "cvc"}
```
Comment (tomtomecek) [inline app/views/users/new.html.haml]: Lines 21 to 27 are a bit wrong - you should not use Rails Helpers - its not benefiting it as you don't want to pass this data to Controller. Sure there is a way around it using these helpers but they are in my opinion more verbose solution, misleading a bit. Why would you create a 'text_field_tag nil ...` I understand your idea - not passing name attr but in this case using pure HTML is more preferred. To make Capybara work with these tags, you also want to make for attribute active for label and have these text field tags an ID. Strongly recommend to fix this based on solution from video.

Hope this helps :)

File: spec/features/friend_invitation_spec.rb [hunk 2]
```diff
@@ -1,5 +1,9 @@
 require 'spec_helper'
 
+Capybara.register_driver :selenium do |app|
+    Capybara::Selenium::Driver.new(app, :browser => :chrome)
+end
```
Comment (tomtomecek) [inline spec/features/friend_invitation_spec.rb]: Changing the driver like that will not work that easily. You have to change RSpec settings and add DatabaseCleaner set up. Also you should label the spec as `js: true`. You will face a lot of feature specs that will test javascript in your apps - its better to have this setup in `spec_helper.rb` or `rails_helper.rb`

File: spec/features/friend_invitation_spec.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 require 'spec_helper'
 
 feature "friend invitation" do
-  scenario "user can invite a friend to join myflix" do
+  scenario "user can invite a friend to join myflix", js: true do
     jane = Fabricate :user
     sign_in_user jane
```
Comment (ppj) [inline spec/features/friend_invitation_spec.rb:6]: @tomtomecek:
Now this call fails to sign `jane` in. I see a `incorrect username or password` flash message (i put a `sleep 10` in there to see that). That means line 5 is not creating `jane` in the test db.
This macro (`sign_in_user`) doesn't fail from other feature specs which are running headless. So the `js: true` is messing it up somehow.

Comment (tomtomecek) [inline spec/features/friend_invitation_spec.rb:6]: Move to the next lesson. There is more things to learn. :)

Comment (ppj) [inline spec/features/friend_invitation_spec.rb:6]: I was a bit uncomfortable finishing this lesson and merging the PR with a failing spec. But will do if you say so :wink:

**Pull request merged**